flowworker/flowworker.py

```diff
@@ -5,30 +5,91 @@
 import sys
 import os
 import shutil
+import time
+from threading import Thread
+from Queue import Queue, Empty
 from scapy.all import *
 
-def packet_hanlder(pkt):
+#MAC Addr of the flow generator 
+FLOWGEN_MAC = "b4:be:b1:6b:00:b5"
+#Timeout const in seconds
+FLOW_TIMEOUT = 10
+
+class Flow:
+    def __init__(self, dst):
+        self.pkts = []
+        self.dst_hash = dst
+        self.last_pkt = time.time()
+    def write(self):
+        if len(self.pkts) == 1:
+            print("Only init pkt recived, flow with dst_hash: {} not written".format(self.dst_hash))
+            return
+
+        i = 0
+        pktdump = PcapWriter(interface+"/"+self.dst_hash, append=True, sync=True)
+        for pkt in self.pkts:
+            pktdump.write(pkt)
+            i += 1
+        pktdump.close()
+        print("Flow with dst_hash: {} written, included {} packets".format(self.dst_hash,i))
+
+
+def packet_handler(pkt):
    #check for ether 
    if type(pkt) is Ether:
        #check dict
        dst = str(pkt.dst)
-       if dst not in dicto:
-           sendp(Ether(dst="b4:be:b1:6b:00:b5", src=dst)/Raw("ENTE"),iface=interface,verbose=False)
-           dicto[dst] = 0
+       if dst not in flows:
+           sendp(Ether(dst=FLOWGEN_MAC, src=dst)/Raw("ENTE"),iface=interface,verbose=False)
```

Comment:
I'd suggest switching src <-> dst mac instead of hardcoding FLOWGEN_MAC. If we only switch mac-addresses we've to specify the FLOWGEN_MAC in the flowgen (dpdk-c) part only. And this script can happily live without actually knowing the variable.